### PR TITLE
[solves #55] Backend/optimization

### PR DESCRIPTION
**Pull request by mathieutrudeau**
- Linked the backend Test project (XUnit) to the main backend
- Wrote DB Connection Tests
- Adjusted the Backend CI to run the Backend Tests from their new location
- Changed some redundant code on the frontend Tests.
- Separated the logic from the controller (Logic in services and endpoints in controllers)
- Tests the Auth service for the Backend

### CODE REVIEW

File: Backend/Backend/Models/User.cs [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
+    /// User email.
+    /// </summary>
     [Required]
     [EmailAddress]
     public string? Email { get; set; }
 
+    /// <summary>
+    /// User password.
+    /// </summary>
     public string? Password { get; set; }
```
Comment (BigSaladMan) [inline Backend/Backend/Models/User.cs:27]: Should probably change this not be a nullable, or at least make it required. Same with Email I think

File: Backend/Backend/Program.cs [hunk 2]
```diff
@@ -94,9 +99,6 @@ public class Program
 
         app.MapControllers();
 
-        // Allows all origins to access the API.
-        app.UseCors(builder=>builder.AllowAnyOrigin().AllowAnyMethod().AllowAnyHeader());
```
Comment (BigSaladMan) [inline Backend/Backend/Program.cs:98]: Is there a reason we swapped this to the localhost version? Feel like hardcoding the localhost into the origins will bite us in the future

**Review by BigSaladMan — APPROVED** [review]

File: Backend/Backend.Tests/GlobalUsings.cs
[the remark this answers is not in the document]
Comment (mathieutrudeau) [reply Backend/Backend.Tests/GlobalUsings.cs:1]: It intends to simplify the code by having all the Usings in the same file.

So say I need to use a library for 10 test files, I'll only have to add it to this file.

File: Backend/Backend/Program.cs (hunk 2 above)
Comment (mathieutrudeau) [reply Backend/Backend/Program.cs:98]: For security measures, it wouldn't make sense to allow the use of our API from any origins. 
For now the localhost version is only for development purposes and we will definitely create a production policy when the time comes.

File: Backend/Backend.Tests/DbContextTests.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    /// <summary>
+    /// Tests that the DbContext can connect to the Database.
+    /// </summary>
+    [Fact]
+    public void DbConnection_Test()
+    {
+        // ARRANGE
+        // Connection Type depends on whether we are running in a container or not
+        string connectionType = Environment.GetEnvironmentVariable("DOTNET_RUNNING_IN_CONTAINER") == "true" ? "DockerConnection" : "DefaultConnection";
```
[the remark this answers is not in the document]
Comment (mathieutrudeau) [reply Backend/Backend.Tests/DbContextTests.cs:38]: Done.

File: Backend/Backend/Models/User.cs (hunk 1 above)
Comment (mathieutrudeau) [reply Backend/Backend/Models/User.cs:27]: Added the password as required. Also added a default constructor to allow for the string properties to be set as not nullable.

**Review by shadijiha — APPROVED** [review]

**Pull request merged**